Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -208,6 +210,32 @@ def check_elisp_root(share, name)
     EOS
   end
 
+  def check_linkage
+    return unless formula.prefix.directory?
+    keg = Keg.new(formula.prefix)
+    checker = LinkageChecker.new(keg, formula)
+    s = ""
+
+    if checker.broken_dylibs?
+      s += <<-EOS.undent
+      Broken dylib links found:
+        #{checker.broken_dylibs.to_a * "\n        "}
+      EOS
+    end
+
+    if checker.undeclared_deps?
+      s += <<-EOS.undent
+      Undeclared dependencies found:
+        #{checker.undeclared_deps * "\n        "}
+
+      Formula is required to declare all direct dependencies.
+      Please check whether these dependencies are needed.
```

Comment:
What's the actionable outcome? Tell people what they need to do here i.e. what lines in the formula need added/removed/changed.